Junimatic/ModConfigMenu.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using StardewModdingAPI;
+using StardewModdingAPI.Events;
+using StardewValley;
+using StardewValley.GameData.BigCraftables;
+using StardewValley.GameData.Objects;
+
+namespace NermNermNerm.Junimatic
+{
+    public class ModConfigMenu
+    {
+        private ModEntry mod = null!;
+
+        public void Entry(ModEntry mod)
+        {
+            this.mod = mod;
+
+            mod.Helper.Events.GameLoop.GameLaunched += this.OnGameLaunched;
+        }
+
+        private void OnGameLaunched(object sender, GameLaunchedEventArgs e)
+        {
+            IManifest ModManifest = this.mod.ModManifest;
+            ModConfig Config = ModEntry.Config;
+
+            // get Generic Mod Config Menu's API (if it's installed)
+            var configMenu = this.mod.Helper.ModRegistry.GetApi<IGenericModConfigMenuApi>("spacechase0.GenericModConfigMenu");
+            if (configMenu is null)
+                return;
+
+            // register mod configs
+            configMenu.Register(
+                mod: ModManifest,
+                reset: () => Config = new ModConfig(),
+                save: () => this.mod.Helper.WriteConfig(Config)
+            );
+
+            configMenu.AddSectionTitle(
+                mod: ModManifest,
+                text: () => "Unlocks"
+            );
+            configMenu.AddBoolOption(
+                mod: ModManifest,
+                name: () => "Unlock Junimo Portal",
```

Comment:
I wonder if users would be confused by this.  E.g. might they click on "Unlock Animal Junimos" below and not this, then wonder why they can't build a network?

Reply:
I agree, it's probably best to combine them all into an "Unlock Everything" command or config.